src/main/software/amazon/event/ruler/JsonRuleCompiler.java

```diff
@@ -608,25 +594,41 @@ private static Patterns processAnythingButListMatchExpression(JsonParser parser)
         return AnythingBut.anythingButMatch(values, hasNumber);
     }
 
-    private static Patterns processAnythingButEqualsIgnoreCaseListMatchExpression(JsonParser parser) throws JsonParseException {
+    private static Patterns processAnythingButValuesSetMatchExpression(JsonParser parser, MatchType matchType)
+            throws JsonParseException {
         JsonToken token;
         Set<String> values = new HashSet<>();
-        boolean hasNumber = false;
         try {
             while ((token = parser.nextToken()) != JsonToken.END_ARRAY) {
                 switch (token) {
                     case VALUE_STRING:
-                        values.add('"' + parser.getText() + '"');
+                        String text = parser.getText();
+                        if (matchType != MatchType.ANYTHING_BUT_IGNORE_CASE && text.isEmpty()) {
+                            barf(parser, "Null prefix/suffix not allowed");
+                        }
+                        values.add((matchType != MatchType.ANYTHING_BUT_SUFFIX ? "\"" : "") +
+                                   text +
+                                   (matchType != MatchType.ANYTHING_BUT_PREFIX ? "\"" : ""));
```

Comment:
thinking long-term, this is an easy place to introduce new bugs with multiple conditions in the same place. Possible to extract this into a method?